### ISSUE TITLE
Repetition of strings after its styled

### ISSUE THREAD

**Issue by Grazty** [tomakita/Colorful.Console#50]
it only started recently i dont know whats wrong i haven't changed anything.
Issue: 
![image](https://user-images.githubusercontent.com/30330990/44055795-0b30df90-9f0c-11e8-881c-3ffcc0277cf4.png)

**Comment by Eregrith** [comment]
I have seen the same problem in my console app but it only happens when I install it through a clickonce setup, not in debug.

**Comment by tomakita** [comment]
Hi @Grazty , 

Thanks for this report.  I agree with @Eregrith -- a reproducible example would help.

**Comment by DoctorLogiq** [comment]
Hi all,
I've just encountered this problem (or a similar one)... when I use Console.WriteLineStyled() and the message string contains the word "Int32", it stops printing (it would seem, after the next whitespace immediately succeeding "Int32") and repeats the line, this time completing it.

I've attached screenshots of the console output.

![engine1](https://user-images.githubusercontent.com/32243802/70486824-1dee1380-1aeb-11ea-8580-e4dae664a0db.png)

![engine2](https://user-images.githubusercontent.com/32243802/70486830-20e90400-1aeb-11ea-8160-51dc7c7b7451.png)

This appears to be affected by the 4th regex added to the styleSheet: `@"\{[\w_]*\}"`, however, it doesn't appear to be the regex's content, changing it to use something other than braces still yields the same results, so it must be something else. The StyleSheet I'm using has 3 other regex's added before this one, and those all work fine. Removing or changing "Int32" to something else makes the issue disappear (as shown in the screenshots above), even though the style is still applied.

**Comment by Eregrith** [comment]
@DoctorLogiq Could you share a code of minimum reproducible example please? I've tried looking through the code but can't find anything that would trigger such a thing

**Comment by Eregrith** [comment]
Ah-ha ! I finally managed to reproduce it and capture the culprit !

It seems these lines are producing a really undesirable effect when one regex is a bit dirty:

https://github.com/tomakita/Colorful.Console/blob/d50ee086ebdafb83141446c5aeff9290dfdb3f1a/src/Colorful.Console/TextAnnotator.cs#L79-L82

In my case, I matched "3c" inside a string of alphanumeric characters through "[0-9]+c", but one style regex happened to end up being "()" (open-close parentheses without anything inside) matching each and every character one after the other. 
Because of that, the TextAnnotator got up to the bit matching the 3c, and then the next location was just the 3 as seen here :
![image](https://user-images.githubusercontent.com/7937994/71833192-bd7fd100-30ac-11ea-8f82-a19d068da2af.png)

So the if got triggered and my start was put to MatchLocation(0,0) which means `vanillaStart` got put to 0, and `vanillaEnd` got put to `currentLocation.Beginning`.
This had the effect to re-paste the whole string from the start up to where it was at just before this loop.
